From 7059848f11c3121a0f6db509d487b0a3d5940221 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Sun, 19 May 2019 13:44:51 +0200
Subject: [PATCH 184/345] MIPS: Handle CPU_CAVIUM_OCTEON3 like
 CPU_CAVIUM_OCTEON2 in clear_page.

Both OCTEON3 and OCTEON2 use the same instrucitons for this.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/mm/page.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/arch/mips/mm/page.c b/arch/mips/mm/page.c
index 78d70c3b1fd2..4ba50631c52b 100644
--- a/arch/mips/mm/page.c
+++ b/arch/mips/mm/page.c
@@ -315,7 +315,8 @@ void build_clear_page(void)
 	memset(labels, 0, sizeof(labels));
 	memset(relocs, 0, sizeof(relocs));
 
-	if (current_cpu_data.cputype == CPU_CAVIUM_OCTEON2) {
+	if (current_cpu_data.cputype == CPU_CAVIUM_OCTEON2 ||
+	    current_cpu_data.cputype == CPU_CAVIUM_OCTEON3) {
 		const unsigned int wb_nudge = 26;
 
 		pg_addiu(&buf, T0, A0, PAGE_SIZE);
@@ -323,25 +324,19 @@ void build_clear_page(void)
 		UASM_i_ADDIU(&buf, A1, A0, 128);
 		uasm_l_clear_pref(&l, buf);
 		uasm_i_zcbt(&buf, A0);
-		uasm_i_pref(&buf, wb_nudge, 0, A0);
 		UASM_i_ADDIU(&buf, A0, A0, 256);
 		uasm_i_zcbt(&buf, A1);
-		uasm_i_pref(&buf, wb_nudge, 0, A1);
 		UASM_i_ADDIU(&buf, A1, A1, 256);
 		uasm_i_zcbt(&buf, A0);
-		uasm_i_pref(&buf, wb_nudge, 0, A0);
 		UASM_i_ADDIU(&buf, A0, A0, 256);
 		uasm_i_zcbt(&buf, A1);
 		uasm_i_pref(&buf, wb_nudge, 0, A1);
 		UASM_i_ADDIU(&buf, A1, A1, 256);
 		uasm_i_zcbt(&buf, A0);
-		uasm_i_pref(&buf, wb_nudge, 0, A0);
 		UASM_i_ADDIU(&buf, A0, A0, 256);
 		uasm_i_zcbt(&buf, A1);
-		uasm_i_pref(&buf, wb_nudge, 0, A1);
 		UASM_i_ADDIU(&buf, A1, A1, 256);
 		uasm_i_zcbt(&buf, A0);
-		uasm_i_pref(&buf, wb_nudge, 0, A0);
 		UASM_i_ADDIU(&buf, A0, A0, 256);
 		uasm_i_zcbt(&buf, A1);
 		uasm_i_pref(&buf, wb_nudge, 0, A1);
-- 
2.25.1

